src/definition/users/IUserGenerate.ts

```diff
@@ -0,0 +1,12 @@
+export interface IUserGenerate {
```

Comment:
I'm not 100% certain about this interface name. What is the main reason for not putting this on the existing user? What about extending the `IUser` interface so that it accepts everything else along with the new fields that are just for when creating a user?

Reply:
@graywolf336 The main reason for not extending the `IUser` is that there are some fields which are non-nullable like `lastLoginAt`, `createdAt` which I felt are to be extended unnecessarily while creating, So I have created a new Interface which is meant for the only generation of a user.

Reply:
sure @graywolf336 I will change that as you suggested :)

Reply:
Awesome, thank you! 😄